addons/website/tools.py

```diff
@@ -212,14 +212,21 @@ def add_form_signature(html_fragment, env_sudo):
             existing_hash_node.getparent().remove(existing_hash_node)
         input_nodes = form.xpath('.//input[contains(@name, "email_")]')
         form_values = {input_node.attrib['name']: input_node for input_node in input_nodes}
+        data_for = html_fragment.xpath('//span[@data-for="contactus_form"]')
+        email_to_value = ''
         # if this form does not send an email, ignore. But at this stage,
         # the value of email_to can still be None in case of default value
         if 'email_to' not in form_values:
             continue
-        elif not form_values['email_to'].attrib.get('value'):
-            form_values['email_to'].attrib['value'] = env_sudo.company.email or ''
+        elif not form_values['email_to'].attrib.get('value')\
+            or (form_values['email_to'].attrib['value'] == 'info@yourcompany.example.com' and len(data_for) > 0):
```

Comment:
- The first leads to a `continue`, the `elif` has thus no importance at all: you can make the code more readable saving `form_values['email_to'].attrib.get('value')` in a variable just after that `if` (before the `elif`), e.g. form_value = ... 
- `len(data_for) > 0` is probably the same as doing `data_for` in python -> the shorter the better

With those two changes, you can probably fit your if on one line (< 100 characters). If not I think a space before the `\` would be better?

However, should you not also check that the data_for element has an email_to value?

Reply:
> However, should you not also check that the data_for element has an email_to value?

@qsm-odoo Ideally yes, but when we compute the `website_form_signature`, the span is not filled with the correct values yet. It still has `t-att-data-values="contactus_form_values"`. As it will be filled later with the company email I thought it was ok like that...